src/ensembl/src/content/app/browser/browserReducer.ts

```diff
@@ -76,6 +76,15 @@ export function browserEntity(
         ...state,
         trackStates: merge({}, state.trackStates, action.payload)
       };
+    case getType(browserActions.deleteGenome): {
+      const genomeIdToRemove = action.payload;
+      const activeGenomeId = state.activeGenomeId;
+      if (activeGenomeId === genomeIdToRemove) {
+        const newState = { ...state, activeGenomeId: null };
+        delete newState.activeEnsObjectIds[activeGenomeId];
```

Comment:
I suppose this is what's producing this error 😄 :

![image](https://user-images.githubusercontent.com/6834224/104188360-0c272000-5411-11eb-87af-f489bd34e613.png)

`{ ...state }` is a shallow copy of the state. Everything that's nested one and more levels deep are the same objects as in the original state. By changing newState.activeEnsObjectIds[activeGenomeId] you are actually mutating the state.

There are several ways out of this predicament. Ideally, we should refactor our browser redux state to use the `createSlice` function from react-toolkit; then we can just pretend that the state is mutable. But in the meantime, you can use either the `pickBy` for the activeEnsObjectIds part of the state or lodash's `cloneDeep` to get a true clone of the state. `pickBy` should be better for performance.

Reply:
Thanks. I have replaced it all with `pickBy`.